### PR TITLE
Allow to extend application FastBoot config via addon hook fastbootConfigTree

### PR DESCRIPTION
**Pull request by SergeAstapov**
Use lodash.merge provided by `ember-cli-lodash-subset` instead of Object.assign for deep merge

Added new app fixture `fastboot-config` that copied from `module-whitelist`.
Implemented `fastbootConfigTree` hook in `test/fixtures/fastboot-config/node_modules/fake-addon/index.js` that extends application config to check for a regression and fix actually works.

### CODE REVIEW

**Comment by SergeAstapov** [comment]
I'm I missing anything so that tests are failing only on Travis only on node@4?

File: test/package-json-test.js [hunk 1]
```diff
@@ -99,15 +99,50 @@ describe('generating package.json', function() {
...
     });
 
-    it('contains additional config from addons', function() {
+    it('contains additional config from ember-fastboot-build-example addon', function() {
       let pkg = fs.readJsonSync(app.filePath('dist/package.json'));
 
       expect(pkg.fastboot.config['foo']).to.equal('bar');
     });
 
+    describe('with addon that implements fastbootConfigTree', function() {
```
Comment (kratiahuja) [inline test/package-json-test.js]: Would recommend moving this after the `'contains app name` test so it is easier to read.

Comment (SergeAstapov) [reply test/package-json-test.js]: @kratiahuja I've moved `'contains app name'` up to make tests for application config grouped.
Thanks for feedback!

**Comment by SergeAstapov** [comment]
@kratiahuja Could you please take a look if updated tests order makes sense to you?

**Pull request merged**